analytics/analytics-ui/src/app/services/analytic-filter.pipe.ts

```diff
@@ -0,0 +1,18 @@
+import { Pipe, PipeTransform } from '@angular/core';
+class Analytic {
```

Comment:
If this class was more representative of the AnalyticDetail class, it could be more widely used throughout the app (such as in the Analytics Service). This gives the benefit of type saftely and makes the code easier to maintain. Furthermore, it should be moved into it's own file.